components/availability/src/Availability.svelte

```diff
@@ -400,12 +592,15 @@
               data-end-time={new Date(slot.end_time).toLocaleString()}
               disabled={slot.availability === AvailabilityStatus.BUSY}
               on:click={() => {
-                slot.selectionStatus =
-                  slot.selectionStatus === SelectionStatus.SELECTED
-                    ? SelectionStatus.UNSELECTED
-                    : SelectionStatus.SELECTED;
-                toggleSelectedTimeSlots(slot);
-                if (!allow_booking) {
+                if (allow_booking) {
+                  slot.selectionStatus =
+                    slot.selectionStatus === SelectionStatus.SELECTED
+                      ? SelectionStatus.UNSELECTED
+                      : slotSelection.length < max_bookable_slots
```

Comment:
A bit of extra logic to make sure that `allow_booking` is true, and that you haven't already selected more slots than you're allowed to

Reply:
Could this logic live in `toggleSelectedTimeSlots` instead?